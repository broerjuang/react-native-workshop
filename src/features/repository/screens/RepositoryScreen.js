// @flow

import React, {Component} from 'react';
import {View} from 'react-native';
import RepoComponent from '../../../components/Repo';
import {SearchBar} from 'react-native-elements';

type Props = {
  total_count: number,
  incomplete_results: boolean,
  items: Array<Items>,
};
type Items = {
  id: number,
  repo_name: string,
  description: string,
  star: number,
  forked: number,
  language: string,
  repoType: string,
};
class RepositoryScreen extends Component<Props, {}> {
  render() {
    let {items = []} = this.props;
    return (
      <View>
        <SearchBar
          onChangeText={(text: string): void => console.log(text)}
          onClearText={() => console.log('remove')}
          placeholderTextColor={'#E1E4EC'}
          placeholder={'Type Here...'}
          icon={{
            type: 'material',
            color: '#909AA3',
            name: 'search',
            style: {alignContent: 'center'},
          }}
          inputStyle={{backgroundColor: '#E1E4EC'}}
          containerStyle={{
            backgroundColor: 'white',
            borderBottomWidth: 0.1,
            borderTopWidth: 0.5,
          }}
        />
        {items.map((item, key) => {
          return (
            <View key={key}>
              <RepoComponent {...item} />
            </View>
          );
        })}
      </View>
    );
  }
}

export default RepositoryScreen;
