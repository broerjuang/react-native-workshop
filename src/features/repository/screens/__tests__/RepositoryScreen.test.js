// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import RepositoryScreen from '../RepositoryScreen';
let datas = {
  // total_count: 2570,
  // incomplete_results: false,
  // items: [
  //   {
  //     id: 2,
  //     repo_name: 'react-navigation',
  //     description: 'Advance react native state management',
  //     star: 2,
  //     forked: 3,
  //     language: 'Java Script',
  //     repoType: 'repo',
  //   },
  //   {
  //     id: 3,
  //     repo_name: 'Kodefox Tax',
  //     description: 'Kodefox daily task',
  //     star: 2,
  //     forked: 3,
  //     language: 'Java Script',
  //     repoType: 'repo',
  //   },
  //   {
  //     id: 4,
  //     repo_name: 'reason',
  //     description: 'Advance react native state management',
  //     star: 2,
  //     forked: 3,
  //     language: 'Ocaml',
  //     repoType: 'forke',
  //   },
  // ],
  navigation: {},
};
describe('Repository screee ', () => {
  it('should render Repository corectly', () => {
    let component = renderer.create(<RepositoryScreen {...datas} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
