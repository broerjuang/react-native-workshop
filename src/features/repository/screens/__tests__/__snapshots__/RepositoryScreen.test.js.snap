// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository screee  should render Repository corectly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#393e42",
            "borderBottomColor": "#000",
            "borderBottomWidth": 1,
            "borderTopColor": "#000",
            "borderTopWidth": 1,
          },
          false,
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 0.1,
            "borderTopWidth": 0.5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onClearText={[Function]}
        placeholder="Type Here..."
        placeholderTextColor="#E1E4EC"
        style={
          Array [
            Object {
              "backgroundColor": "#303337",
              "borderRadius": 3,
              "color": "#86939e",
              "fontSize": 17.5,
              "height": 30,
              "margin": 8,
              "overflow": "hidden",
              "paddingLeft": 26,
              "paddingRight": 19,
            },
            false,
            false,
            false,
            Object {
              "backgroundColor": "#E1E4EC",
            },
            undefined,
            false,
          ]
        }
        underlineColorAndroid="transparent"
      />
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#909AA3",
              "fontSize": 16,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "position": "absolute",
                "top": 15.5,
              },
              Object {
                "left": 16,
              },
              Object {
                "alignContent": "center",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î¢¶
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderWidth": 0.2,
            "flexDirection": "row",
            "padding": 6,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 5,
              "flexDirection": "column",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
              }
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginRight": 15,
                "marginVertical": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                0
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                0
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginHorizontal": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
